lib/services/elasticsearch.js

```diff
@@ -703,17 +703,90 @@ function refreshIndexIfNeeded(data, response) {
 }
 
 function formatESError(error) {
-  var kuzzleError = new Error(error.message);
+  var
+    formattedMessages,
+    message,
+    kuzzleError;
+
+  formattedMessages = {
```

Comment:
Having the regexes expressed as strings and evaluated on each function likely forces the code to compile the regexes on each call.
Would be slightly more efficient using an array of some pre-compiled regexes on the module level.

Something like:

```js
function Elasticsearch (kuzzle, options, config) {
  this.errorMessagesMapping = [
    {   regex: /^\[illegal_argument_exception\] object mapping \[(.*?)\] can't be changed from nested to non-nested$/,
        replacement: 'Cannot change mapping for field "$1" from nested to another type'   },
    {   regex: /^\[illegal_argument_exception\] object mapping \[(.*?)\] can't be changed from non-nested to nested$/,
        replacement: 'Cannot change mapping for field "$1" from non-nested to nested' },
    [..]
  ];
}

Elasticsearch.prototype.formatESError (error) {
  var message = error.message;

  this.errorMessagesMapping.some(mapping => {
    message = message.replace(mapping.regex, mapping.replacement);
    return (message !== error.message);
  });

  [..]
}

module.exports = Elasticsearch;